sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentProducer.java

```diff
@@ -147,7 +147,9 @@ public DocumentProducer(
 
         this.correlatedActivityId = correlatedActivityId;
 
-        this.cosmosQueryRequestOptions = cosmosQueryRequestOptions != null ? cosmosQueryRequestOptions : new CosmosQueryRequestOptions();
+        this.cosmosQueryRequestOptions = cosmosQueryRequestOptions != null ?
+                                             ModelBridgeInternal.createQueryRequestOptions(cosmosQueryRequestOptions)
```

Comment:
I remember this was fixed earlier. Did we undo this somehow?

Reply:
Yes. Got lost in the v4->master merge as discussed offline.

Reply:
@mbhaskar  @moderakh  - I checked the PR logs - and this PR was never created against V4. 
It was only fixed on v2 and v3 SDK, not on V4 SDK - https://github.com/Azure/azure-sdk-for-java/pulls?q=is%3Apr+author%3Ambhaskar+is%3Aclosed

Reply:
Also, V4 was merged into master on this PR - https://github.com/Azure/azure-sdk-for-java/pull/8351
So this had to be manually ported to V4 if it had to go into V4. 
@mbhaskar  - please check if you had any other PRs that we didn't port to V4, we should do that now.